doc/en/docs/dependencies.md

```diff
@@ -0,0 +1,79 @@
+# Dependent library installation
+
+## Windows
+
+This section is used to compile and install the dependent libraries under
+windows system from source codes. The ONLY compiler used is Visual Studio 2015.
+If you intend to generate a 32-bit/64-bit singa solution, please configure all the
+VS projects for the dependent libraries as 32-bit/64-bit. This can be done by
+"Configuration Manager" in VS 2015 or use coresponding generator in cmake. When compiling the following libraries, you
+may get system-specific warnings/errors. Please fix them according to the
+prompts by VS.
+
+### Google Logging
+
+The glog library is an optional library for singa project. But it is needed for
+protobuf check on cmake. Since the latest release version of glog will encounter error C2084 on sprintf function
+under VS2015, we test the compilation and installation with the
+latest codes of glog. The code is from [git](https://github.com/google/glog). 
+
+Step 1: Download and decompress the source code. Or use `git clone
+https://github.com/google/glog` to get the code.
+
+Step 2: Open "glog.sln" file under project folder. You will get a conversion
+dialog and please finish it by the prompts. Compile all the projects in the solution after
+proper configuration, especially "libglog" and "libglog_static" projects. 
+
+Step 3: Copy all the header files and the entire directory named "glog" under
+"src\windows\" folder into the installation include folder (or system folder).
+Copy all the generated library files into the installation library folder (or
+system folder).
+
+Step 4: Done.
+
+### Google protobuf 
+
+Tested on version 2.6.1:
+
+Step 1: Download and decompress the source code.
+
+Step 2: Open "protobuf.sln" file under "vsprojects" folder. You will get a conversion
+dialog and please finish it by the prompts. Compile all the projects in the solution after proper
+configuration. Especially "libprotobuf", "libprotobuf-lite", "libprotoc" and
+"protoc" projects.
+
```

Comment:
Solution: Right click on libprotobuf project -> Properties -> C/C++ -> Preprocessor -> Add "_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS" to Preprocessor Definitions.